Documentation/ceph-bucket-notification-crd.md

```diff
@@ -0,0 +1,56 @@
+# Ceph bucket notifications CRD
+
+## Overview
+
+In its new release, Ceph introduced the bucket notifications feature. It allows to send messages to various endpoints when a new event occurs on a bucket (ref: <https://docs.ceph.com/docs/master/radosgw/notifications/)>
+
+Setup of those notifications are normally done by sending HTTP requests to the RGW, either to create/delete topics pointing to specific endpoints, or create/delete bucket notifications based on those topics.
+
+This functionality eases this process by avoiding to use external tools or scripts. It is replaced by creation of CR definitions that  contain all the information necessary to create topics and/or notifications, which the rook operator processes.
+
+## Goals
+
+Creates a CRD for topics and a CRD for notifications, defining all the necessary and optional information for the various endpoints.
+
+Extends the rook operator to handle the CRs that would be submitted by users.
+
+## Implementation
+
+The CR for a topic configuration takes this form:
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: Topic
+metadata:
+  name: rook-ceph
+  namespace: rook-ceph
+Spec:
+  pushEndpoint: Kafka/HTTP/RabbitMQ, mandatory
```

Comment:
we probably need a different spec per endpoint type. they contain different fields (with some overlap)

Reply:
Do we? In the POST configuration, referring to the doc @attribut 8, it's a single field, "push-endpoint", with values starting by http(s)|kafka|amqp. That's why we put a single field here to.

Reply:
I don't think that packing the parameters into the URL is a clear way to store that info.
in the docs we use it in a the older "pubsub" mode.
here, we should follow:
https://docs.ceph.com/docs/master/radosgw/notifications/#create-a-topic

Reply:
So something like:
```
Spec:
  httpPushEndpoint: xyz, optional
  kafkaPushEndpoint: xyz, optional
  amqpPushEndpoint: xyz, optional
```